docs/advanced/exceptions.rst

```diff
@@ -43,15 +47,23 @@ at its exception handler.
 |                                      | of bounds access in ``__getitem__``, |
 |                                      | ``__setitem__``, etc.)               |
 +--------------------------------------+--------------------------------------+
-| :class:`pybind11::value_error`       | ``ValueError`` (used to indicate     |
-|                                      | wrong value passed in                |
-|                                      | ``container.remove(...)``)           |
-+--------------------------------------+--------------------------------------+
 | :class:`pybind11::key_error`         | ``KeyError`` (used to indicate out   |
 |                                      | of bounds access in ``__getitem__``, |
 |                                      | ``__setitem__`` in dict-like         |
 |                                      | objects, etc.)                       |
 +--------------------------------------+--------------------------------------+
+| :class:`pybind11::value_error`       | ``ValueError`` (used to indicate     |
+|                                      | wrong value passed in                |
+|                                      | ``container.remove(...)``)           |
++--------------------------------------+--------------------------------------+
+| :class:`pybind11::type_error`        | ``TypeError``                        |
++--------------------------------------+--------------------------------------+
+| :class:`pybind11::buffer_error`      | ``BufferError``                      |
++--------------------------------------+--------------------------------------+
+| :class:`pybind11::import_error`      | ``import_error``                     |
++--------------------------------------+--------------------------------------+
+| Any other exception                  | ``RuntimeError``                     |
```

Comment:
I don't think this last line is correct. Especially considering that the table starts with `std::exception`. Really, it's "any other exception inheriting from `std::exception`". I don't think pybind11 would handle `throw 5;` or `throw boost::exception{whateveer};`.

Reply:
It does though, I think, @bstaletic:

https://github.com/pybind/pybind11/blob/17c22b9e0f56276f212b44f6cf0e6846a01623b2/include/pybind11/detail/internals.h#L239-L242

Reply:
Fair enough. Forget what I said.

Reply:
(This comment is for me an argument in favour of being able to find the code for an advanced reader, as the ground truth is the code).

Reply:
To me, that's an argument for adding correct documentation. You don't want to dive into the code every time, right?
This is implementation. And of course the code is the ground truth, but you don't just say "let's give up on docs and always read the code", right?

Reply:
Anyway, that's what I think. If you can convince other reviewers, why not, but I'm definitely not convinced of those links. Very happy to see the table completed, though!